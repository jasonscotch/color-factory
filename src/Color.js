import { Link } from 'react-router-dom';
import "./Color.css";


const Color = ({ hex, color }) => {
    const style = {
        backgroundColor: hex
    };

  return (
    <div className="Color" style={style}>
      <h2>This is {color}!</h2>
      <p>
        <Link to="/colors">Go Back</Link>
      </p>
    </div>
  );
}

export default Color;
