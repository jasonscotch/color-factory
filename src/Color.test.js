import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from 'react-router-dom';
import Color from "./Color";

it("renders without crashing", function () {
  render(
  <MemoryRouter>
    <Color />
  </MemoryRouter>
  
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
        <Color />
    </MemoryRouter>
);
  expect(asFragment()).toMatchSnapshot();
});