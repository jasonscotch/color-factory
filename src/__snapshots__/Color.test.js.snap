// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Color"
  >
    <h2>
      This is !
    </h2>
    <p>
      <a
        href="/colors"
      >
        Go Back
      </a>
    </p>
  </div>
</DocumentFragment>
`;
